components/centraldashboard-angular/frontend/src/app/services/namespace.service.ts

```diff
@@ -0,0 +1,201 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { ReplaySubject } from 'rxjs';
+import { filter, map, switchMap, take } from 'rxjs/operators';
+import { getQueryParams, getUrlFragment } from '../shared/utils';
+import { Namespace } from '../types/namespace';
+import { EnvironmentService } from './environment.service';
+import { LocalStorageService } from './local-storage.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CDBNamespaceService {
+  private _namespacesSubject = new ReplaySubject<Namespace[]>(1);
+  private _currentNamespaceSubject = new ReplaySubject<Namespace>(1);
+
+  namespacesSubject = this._namespacesSubject.asObservable();
+  currentNamespaceSubject = this._currentNamespaceSubject.asObservable();
+
+  private user: string;
+  private currentNamespace: Namespace | undefined;
```

Comment:
Maybe we can do this:
```suggestion
  namespaces = this._namespacesSubject.asObservable();
  currentNamespace = this._currentNamespaceSubject.asObservable();

  private _user: string;
  private _currentNamespace: Namespace | undefined;
```
wdyt @orfeas-k ?

Reply:
Sure, but I 'll rename them from `_variable` to `prvVariable` because that's what we do the in the rest of KF WAs as well.